Src/Test.UnitTests.Sarif.PatternMatcher.Cli/AnalyzeCommandTests.cs

```diff
@@ -277,6 +280,98 @@ public void AnalyzeCommand_ShouldThrowExceptionForConflictingFileSizeOptions()
         }
 
+
+        [Fact]
+        public void AnalyzeCommand_FailureLevelShouldBeDefaultUnlessDynamicallyValidated()
+        {
+            var sb = new StringBuilder();
+
+            var testCases = new[] {
+                new {
+                    jsonLevel = FailureLevel.Note,
+                    dynamicValidationEnabled = false,
+                    expectedNoValidation = FailureLevel.Note,
+                    expectedStaticOnly = FailureLevel.Note,
+                    expectedStaticDynamic = FailureLevel.Warning
+                },
+                new {
+                    jsonLevel = FailureLevel.Warning,
+                    dynamicValidationEnabled = false,
+                    expectedNoValidation = FailureLevel.Warning,
+                    expectedStaticOnly = FailureLevel.Warning,
+                    expectedStaticDynamic = FailureLevel.Warning
+                },
+                new {
+                    jsonLevel = FailureLevel.Error,
+                    dynamicValidationEnabled = false,
+                    expectedNoValidation = FailureLevel.Error,
+                    expectedStaticOnly = FailureLevel.Error,
+                    expectedStaticDynamic = FailureLevel.Warning
+                },
+                new {
+                    jsonLevel = FailureLevel.Note,
+                    dynamicValidationEnabled = true,
+                    expectedNoValidation = FailureLevel.Note,
+                    expectedStaticOnly = FailureLevel.Note,
+                    expectedStaticDynamic = FailureLevel.Error
+                },
+                new {
+                    jsonLevel = FailureLevel.Warning,
+                    dynamicValidationEnabled = true,
+                    expectedNoValidation = FailureLevel.Warning,
+                    expectedStaticOnly = FailureLevel.Warning,
+                    expectedStaticDynamic = FailureLevel.Error
+                },
+                new {
+                    jsonLevel = FailureLevel.Error,
+                    dynamicValidationEnabled = true,
+                    expectedNoValidation = FailureLevel.Error,
+                    expectedStaticOnly = FailureLevel.Error,
+                    expectedStaticDynamic = FailureLevel.Error
+                }
+            };
+
+            foreach (var testCase in testCases)
+            {
+                string definitionsText = GetSingleLineRuleDefinitionFailureLevel(testCase.jsonLevel);
+                string testScenarioName = "NoValidation";
+                SarifLog logFile = RunAnalyzeCommandWithDynamicValidation(definitionsText,
+                                                                          testScenarioName,
+                                                                          testCase.dynamicValidationEnabled);
+
+                ValidateFailureLevelByValidationType(expectedFailureLevel: testCase.expectedNoValidation,
+                                                     sarifLog: logFile,
```

Comment:
>sarifLog: logFile [](http://example.com/codeflow?start=53&length=17)

same note. does these duplicated names, 'sarifLog: logfile' help use? can you collapse them?

for example, what if you named your local variable 'sarifLog'?

in general, your goal should be for both your parameter names in callees and call sites to be sufficiently descriptive to be standalone. with the result that you do *not* have to provide a named argument in order for the code to be readable. there's certainly no excuse for that when you own both the variable names and the called methods, as here. :)

the named arguments are most helpful when providing standalone literals and when wiring up other data where you don't own the name...